go/vt/mysqlctl/backupengine.go

```diff
@@ -138,23 +142,30 @@ type BackupManifest struct {
 // FindBackupToRestore returns a selected candidate backup to be restored.
 // It returns the most recent backup that is complete, meaning it has a valid
 // MANIFEST file.
-func FindBackupToRestore(ctx context.Context, cnf *Mycnf, mysqld MysqlDaemon, logger logutil.Logger, dir string, bhs []backupstorage.BackupHandle) (backupstorage.BackupHandle, error) {
+func FindBackupToRestore(ctx context.Context, cnf *Mycnf, mysqld MysqlDaemon, logger logutil.Logger, dir string, bhs []backupstorage.BackupHandle, snapshotTime time.Time) (backupstorage.BackupHandle, error) {
 	var bh backupstorage.BackupHandle
 	var index int
+	unixZeroTime := time.Time{}
 
 	for index = len(bhs) - 1; index >= 0; index-- {
 		bh = bhs[index]
 		// Check that the backup MANIFEST exists and can be successfully decoded.
-		_, err := GetBackupManifest(ctx, bh)
+		bm, err := GetBackupManifest(ctx, bh)
 		if err != nil {
 			log.Warningf("Possibly incomplete backup %v in directory %v on BackupStorage: can't read MANIFEST: %v)", bh.Name(), dir, err)
 			continue
 		}
 
-		logger.Infof("Restore: found backup %v %v to restore", bh.Directory(), bh.Name())
-		break
+		backupTime, _ := time.Parse(time.RFC3339, bm.BackupTime)
```

Comment:
We should handle this error, but we do need to be careful about backward compatibility. I think we should only call `time.Parse()` if `bm.BackupTime != ""`, and if it's set we should require the time to be valid.